app/database/user_store.go

```diff
@@ -16,7 +17,7 @@ func (s *UserStore) ByID(id int64) *DB {
 
 const deleteWithTrapsBatchSize = 1000
 
-// DeleteTemporaryWithTraps deletes temporary users who don't have active sessions.
+// DeleteTemporaryWithTraps deletes temporary users who don't have sessions or whose sessions expired more than `delay` ago.
```

Comment:
It means if we apply session cleanup (for whatever reason), we may delete temp users much earlier than expected.  Same if they logout (even if technically there is no interface for logout)
Hum I think that will not work, we need to find another criteria...
I would say that if it has no session, we use the `latest_login_at` value (or `latest_activity_at` but at the condition it is updated when we ask for token extension (probably not the case for now)).

Reply:
1. As you said, we don't have the "log out" feature which could delete sessions
2. When we delete old sessions (in createAccessToken()) we still keep 10 most recent sessions, so this will not affect deletion of temporary users.
So, I think the current solution will work.

Reply:
Also, we do not update `latest_login_at` on access token refreshing (as that's not considered as new logging in).
For `latest_activity_at`, it is only updated in itemGetHintToken, itemGetAnswerToken, accessTokenCreate (but not for temporary users), userDataRefresh, itemTaskTokenGenerate, so not enough to rely on this column.

Reply:
> When we delete old sessions (in createAccessToken()) we still keep 10 most recent sessions, so this will not affect deletion of temporary users.

But that's a temporary assumption. We could choose to completely delete expired session... and if the new session is created afterwards and fails, we may have no sessions.